### ISSUE TITLE
Bug: renaming or deleting internal files leaves junk java files in src

### ISSUE THREAD

**Issue by rafcolson** [GoranM/bdx#902]
I'm experiencing some problems when renaming or deleting internal files. You wouldn't expect internal files to exist internally. Especially as leftovers when renaming or deleting those.

**Comment by GoranM** [comment]
I think there was a reason why we didn't do that already ... Maybe @SolarLune remembers.

**Comment by SolarLune** [comment]
Yeah, we used to delete those internal files made external, but stopped when multi-blend exporting came around (as text resources could overwrite text resources exported from other blend files). 

I think any exported text files need to be stored in `/src` when running the game in order for them to be detected by the build process, and we can't even delete them after running, as you can run the game without export now.

**Comment by rafcolson** [comment]
>I think any exported text files need to be stored in `/src` when running the game in order for them to be detected by the build process, and we can't even delete them after running, as you can run the game without export now.

Couldn't we add a temporary folder `/__internal` inside `/src` for internal java files?

**Comment by SolarLune** [comment]
> Couldn't we add a temporary folder /__internal inside /src for internal java files?

Yeah, I guess that could work.

> The ut.save_internal_java_files(ut.src_root()) call in exprun.py won't overwrite texts which already exist, so I don't understand what the problem is exactly.

Ah, I thought it did - I guess I was mistaken. But we still couldn't wipe that folder before every export because the folder could contain text files from blend files not exported this time around (i.e. there's a class in "Player.blend" that gets exported at first along with "Game.blend", but afterwards only "Game.blend" is exported and run).

**Comment by rafcolson** [comment]
@SolarLune @GoranM 

Thinking about this issue some more, a solution could be to add another check-box in the Render tab in Blender: _Synchronize Internal Text Files_, which is set to `True` by default. With the internal text files now residing in a separate folder in the Project's source folder: ___internal_. In fact, we could also use this folder for the _inst_ folder and `BdxApp.java`.

So what do you guys think about this suggestion?

**Comment by SolarLune** [comment]
That sounds fine to me. People rarely (if ever) need to mess with BDXApp.java.

**Comment by GoranM** [comment]
Ok.
